### PR TITLE
Lea 529 - migrate polkadot

### PR DESCRIPTION
**Pull request by zurgl**
migrate polkadot to the new reducer

### CODE REVIEW

File: components/protocols/polkadot/lib/index.ts [hunk 1]
```diff
@@ -1,24 +1,13 @@
...
-      ? `https://${process.env.DATAHUB_POLKADOT_WESTEND_RPC_URL}/apikey/${process.env.DATAHUB_POLKADOT_API_KEY}`
-      : `wss://${process.env.DATAHUB_POLKADOT_WESTEND_WS_URL}/apikey/${process.env.DATAHUB_POLKADOT_API_KEY}`;
+export const getNodeUrl = (network: string) => {
+  if (network === 'Datahub') {
+    return `${process.env.POLKADOT_DATAHUB_URL}/apikey/${process.env.DATAHUB_POLKADOT_API_KEY}`;
   } else {
-    return protocol === POLKADOT_PROTOCOLS.RPC
-      ? `https://${process.env.DATAHUB_POLKADOT_MAINNET_RPC_URL}/apikey/${process.env.DATAHUB_POLKADOT_API_KEY}`
-      : `wss://${process.env.DATAHUB_POLKADOT_MAINNET_WS_URL}/apikey/${process.env.DATAHUB_POLKADOT_API_KEY}`;
+    return 'wss://westend-rpc.polkadot.io';
```
Comment (pyr0gan) [inline components/protocols/polkadot/lib/index.ts:5]: Could we use `process.env.POLKADOT_TESTNET_URL` here?

File: pages/api/polkadot/connect.ts [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-import {getSafeUrl} from '@figment-polkadot/lib';
+import {getNodeUrl} from '@figment-polkadot/lib';
 
 export default async function connect(
-  _req: NextApiRequest,
+  req: NextApiRequest,
   res: NextApiResponse<string>,
 ) {
-  let provider;
+  let provider: any;
```
Comment (pyr0gan) [inline pages/api/polkadot/connect.ts:10]: `let provider: any;` here prevents the TS warning on `.disconnect()` 👍

**Review by pyr0gan — APPROVED** [review]
Tested in Gitpod, works 100%

Changes look good 🚀

File: components/protocols/polkadot/lib/index.ts (hunk 1 above)
Comment (zurgl) [reply components/protocols/polkadot/lib/index.ts:5]: planned for later

**Pull request merged**